### PR TITLE
feat(tests): add some code snippets that should be detected

### PR DESCRIPTION
**Pull request by vlaurier**
Some functional tests to ensure the library triggers some banned code.

### CODE REVIEW

**Comment by szepeviktor** [comment]
Very nice thought!

**Comment by szepeviktor** [comment]
Is `define()` banned?

**Comment by vlaurier** [comment]
> Is `define()` banned?

By default it is not. But you can configure the library as you wish. Have a look at the end of the README for this.

File: Makefile [hunk 1]
```diff
@@ -25,8 +25,17 @@ app-security-check: ## to check if any security issues in the PHP dependencies
 app-static-analysis: ## to run static analysis
 	php -dmemory_limit=-1 vendor/bin/phpstan analyze src tests -l 7
 
+files := $(wildcard snippets/*)
 app-test: ## to run unit tests
 	vendor/bin/phpunit
+	@for filename in $(files); do \
```
Comment (mremi) [inline Makefile]:
```suggestion
	@for filename in $$(find snippets -type f -name *.php); do \
```
so no need anymore `files`

File: snippets/exit.php [hunk 2]
```diff
@@ -0,0 +1,4 @@
+<?php
+
+exit;
```
Comment (mremi) [inline snippets/exit.php]: please create die.php as well

File: Makefile [hunk 3]
```diff
@@ -27,6 +27,14 @@ app-static-analysis: ## to run static analysis
 
 app-test: ## to run unit tests
 	vendor/bin/phpunit
+	@for filename in $$(find snippets -type f -name *.php); do \
```
Comment (mremi) [inline Makefile]: should be in a dedicated target, because not called during CI here (Travis calls `make ci`)

File: Makefile [hunk 4]
```diff
@@ -28,8 +28,18 @@ app-static-analysis: ## to run static analysis
 app-test: ## to run unit tests
 	vendor/bin/phpunit
 
+app-functional-test: ## test some code snippets are detected as banned code
```
Comment (mremi) [inline Makefile]: please change to `app-test-functional` to have the prefix `app-test`

**Pull request merged**